doc/source/tutorials/filters.rst

```diff
@@ -0,0 +1,372 @@
+memory such that the resulting array of bytes will contain each element's first
+byte, followed by each element's second byte, etc. After shuffling the bytes
+would therefore be:
+
+.. code-block:: none
+
+    0x01 0x02 0x03 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00
+
+Note the longer run of zero-valued bytes, which will compress more efficiently.
+
+The byteshuffle filter does not support any options.
+
+Bitshuffle
+~~~~~~~~~~
+
+The filter ``TILEDB_FILTER_BITSHUFFLE`` performs
+`bit shuffling <https://www.sciencedirect.com/science/article/pii/S2213133715000694>`_
+of data as a way to improve compression ratios. The bitshuffle implementation
+used in TileDB comes from `<https://github.com/kiyo-masui/bitshuffle>`_.
+
+Bitshuffling is conceptually very similar to byteshuffling, but operates on the
+bit granularity rather than the byte granularity. Shuffling at the bit level
+can increase compression ratios even further than the byteshuffle filter.
+
+Typically this filter would not be used on its own, but rather immediately
+followed by a compression filter in a filter list.
+
+Positive-delta encoding
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The filter ``TILEDB_FILTER_POSITIVE_DELTA`` performs positive-delta encoding.
+Positive-delta encoding is a form of delta encoding that only works when the
+delta value is positive. Positive-delta encoding can result in better
+compression ratios on the encoded data. Typically this filter would not be used
+on its own, but rather immediately followed by a compression filter in a filter
+list.
+
+For example, if the data being filtered was the sequence of integers ``100, 104,
+108, 112, ...``, then the resulting positive-encoded data would be ``0, 4, 4,
+4, ...``. This encoding is advantageous in that producing long runs of repeated
+values can result in better compression ratios, if a compression filter is added
+after positive-delta encoding.
+
+The filter operates on a "window" of values at a time, which can help in some
+cases to produce longer runs of repeated delta values.
+
+The positive-delta encoding filter supports one option:
+
+* ``TILEDB_POSITIVE_DELTA_MAX_WINDOW`` (type ``uint32_t``): The window size in
+  bytes to use. Default: 1024.
+
+.. note::
+
+    Positive-delta encoding is particularly useful for the offsets of
+    variable-length attribute data, which by definition will always have
+    positive deltas. The above example of the form ``100, 104, 108, 112`` can
+    easily arise in the offsets, if for example you have a variable-length
+    attribute of 4-byte values with mostly single values per cell instead of a
+    variable number.
+
+
+Bit width reduction
+~~~~~~~~~~~~~~~~~~~
+
+The filter ``TILEDB_FILTER_BIT_WIDTH_REDUCTION`` performs bit-width reduction,
+which is a form of compression.
+
+Bit-width reduction examines a window of attribute values, and determines if all
+of the values in the window can be represented by a narrower datatype. If so,
```

Comment:
you define what a narrower datatype is below, but maybe just say a `datatype of smaller bytesize`?